spec/controllers/staff_controller_spec.rb

```diff
@@ -0,0 +1,121 @@
+# spec/controllers/staff_controller_spec.rb
+require 'rails_helper'
+
+RSpec.describe StaffController, type: :controller do
+  describe "GET #index" do
+    before(:each) { @response = get :index }
+    it "renders the template" do
+      expect(@response).to render_template(:index)
+    end
+    it "loads all of the staff into @staff" do
+      expect(@response.body.length).to be_truthy
+    end
+  end
+
+  describe "GET #show" do
+    before(:each) do
+      @staff = create(:staff)
+      @response = get :show, params: { id: @staff }
+    end
+
+    it "renders the template" do
+      expect(@response).to render_template(:show)
+    end
+    it "loads the staff into @staff" do
+      expect(assigns(:staff)).to eq(@staff)
+    end
+  end
+
+  describe "GET #new" do
+    it "renders the template" do
+      get :new
+      expect(@response).to render_template(:new)
+    end
+  end
+
+  describe "GET #edit" do
+    before(:each) do
+      @staff = create(:staff)
+      @response = get :edit, params: { id: @staff }
+    end
+
+    it "renders the template" do
+      expect(@response).to render_template(:edit)
+    end
+    it "loads the staff into @staff" do
+      expect(assigns(:staff)).to eq(@staff)
+    end
+  end
+
+  describe "POST #create" do
+    context "with valid attributes" do
+      before(:each) { @person = create(:person) }
+      it "changes staff count by +1" do
+        expect { 
+          post :create, params: { staff: FactoryGirl.attributes_for(:staff, person_id: @person.id) }
+        }.to change(Staff, :count).by(1)
+      end
+      it "loads the staff into @staff" do
+        post :create, params: { staff: FactoryGirl.attributes_for(:staff, person_id: @person.id) }
+        expect(response).to redirect_to(Staff.last)
+      end
+    end
+
+    context "with invalid attributes" do
+      it "does not save the new staff" do
+        expect {
+          post :create, params: { staff: FactoryGirl.attributes_for(:staff, :invalid) }
+        }.to_not change(Staff, :count)
+      end
+      it "re-runs staff#new" do
+        post :create, params: { staff: FactoryGirl.attributes_for(:staff, :invalid) }
+        expect(response).to render_template(:new)
+      end
+    end
+  end
+
+  describe "PATCH #update" do
+    before(:each) { @staff = create(:staff, pod: 2) }
```

Comment:
I'd love for these tests to utilize `subject`. We could rewrite some of these like this:

```ruby
describe "PATCH #update" do
  subject { put :update, params: { id: @staff, staff: params }

  let(:params) {  { FactoryGirl.attributes_for(:person) } }

  it "blah blah" do
    subject
    expect(assigns(:staff)).to eq(@staff)
  end

  context "with invalid attributes" do
    let(:params) { { pod: nil } }

    # more tests
  end
end
```